Data.R

```diff
@@ -1,97 +1,131 @@
+dat0 %>% rename(Country = `Country/Region`) %>% head
+#' `mutate` adds new columns that are functions of old columns
+dat0 %>% mutate(DeathRate = Deaths / Confirmed) %>% head
 dat0 %>% 
   mutate(DeathRate = Deaths / Confirmed) %>% slice_tail(n=20)
-# Transmute lets you keep the new variable
-dat0 %>% transmute(DeathRate = Deaths / Confirmed)
-# Relocate changes column order
-dat0 %>% relocate(Confirmed:Active, .before = `Province/State`)
-# Summarise collapses a data frame to a single row
+#' `transmute` lets you keep the new variable
+dat0 %>% transmute(DeathRate = Deaths / Confirmed) %>% head
+#' `relocate` changes column order
+dat0 %>% relocate(Confirmed:Active, .before = `Province/State`) %>% head
+#' `summarise` collapses a data frame to a single row
 dat0 %>% summarise(Confirmed = mean(Confirmed, na.rm = TRUE))
-# can use pipe(%>%) to string multiple commands together
-dat0 %>% group_by(`Country/Region`) %>% select(Confirmed:Deaths) %>% summarise( Confirmed = mean(Confirmed, na.rm = TRUE), Deaths = mean(Deaths, na.rm = TRUE))
-#How to concatenate Strings
-paste(month.name)    
-paste(month.abb, month.name)
-# sep allows you to add in things between the strings
-paste(month.abb, "1", month.name, sep = ";")
+#' can use pipe (`%>%`) to string multiple commands together
+dat0 %>% group_by(`Country/Region`) %>% select(Confirmed:Deaths) %>% 
+  summarise( Confirmed = mean(Confirmed, na.rm = TRUE)
+             , Deaths = mean(Deaths, na.rm = TRUE)) %>% head
+#' How to concatenate Strings
+paste(month.name) %>% pander
+paste(month.abb, month.name) %>% pander
+#' sep allows you to add in things between the strings
+paste(month.abb, "1", month.name, sep = ";") %>% pander
 
-paste(month.abb, month.name, collapse = "','")
-# \n and \t add newline and tab respectively, theyre escape keys
-# cat is useful to show messages on the screen, message is almost identical to cat except its easier to filter
+paste(month.abb, month.name, collapse = "','") %>% pander
+#' `\n` and `\t` add newline and tab respectively, theyre escape keys
+#' `cat` is useful to show messages on the screen, `message` is almost identical 
+#' to `cat` except its easier to filter
 paste(month.abb, month.name, collapse = "\t") %>% cat 
-dat0<- mutate(dat0, Country = paste0(`Country/Region`,", ", paste0(`Country/Region`, ifelse(`Province/State` == "", "", paste(",", `Province/State`)))))
 
+#' Create a new column called `Country` out of `Country/Region` and 
+#' `Province/State` then count up the frequency of each value:
+#' 
+dat0<- mutate(dat0, Country = paste0(`Country/Region`,", ", paste0(`Country/Region`, ifelse(`Province/State` == "", "", paste(",", `Province/State`)))))
+table(dat0$Country) %>% cbind
 
-table(dat0$Country)
 #dat0[,'FOO']
 #dat0[['FOO']]
-{length(letters)}
-sqrt('a')
-sqrt({'a'; 4})
-sqrt({ls(all=T); 4})
-sqrt({print(ls(all=T)); 4})
+# {length(letters)}
+# sqrt('a')
+# sqrt({'a'; 4})
+# sqrt({ls(all=T); 4})
+# sqrt({print(ls(all=T)); 4})
```

Comment:
Some of these in-class examples were supposed to throw errors to make a point, but here I commented them out so they don't throw errors when running this script.